.azure-pipelines/ci-build.yml

```diff
@@ -2,354 +2,344 @@
-            - powershell: |
-                $fileNames = "$(Pipeline.Workspace)/Nugets/Microsoft.OpenApi.Hidi.*.nupkg", "$(Pipeline.Workspace)/Nugets/Microsoft.OpenApi.Readers.*.nupkg", "$(Pipeline.Workspace)/Nugets/Microsoft.OpenApi.Workbench.*.nupkg"
-                foreach($fileName in $fileNames) {
-                  if(Test-Path $fileName) {
-                    rm $fileName -Verbose
+        
+        # publish hidi as an .exe
+        - task: DotNetCoreCLI@2
+          displayName: publish Hidi as executable
+          inputs:
+            command: 'publish'
+            arguments: -c Release --runtime win-x64 /p:PublishSingleFile=true /p:PackAsTool=false --self-contained --output $(Build.ArtifactStagingDirectory)/Microsoft.OpenApi.Hidi-v$(hidiversion)
+            projects: 'src/Microsoft.OpenApi.Hidi/Microsoft.OpenApi.Hidi.csproj'
+            publishWebProjects: False
+            zipAfterPublish: false
+            
+        - task: CopyFiles@2
+          displayName: Prepare staging folder for upload
+          inputs:
+            targetFolder: $(Build.ArtifactStagingDirectory)/Nugets
+            sourceFolder: $(Build.ArtifactStagingDirectory)
+            content: '*.nupkg'
+            
+    - stage: deploy
+      condition: and(contains(variables['build.sourceBranch'], 'refs/heads/master'), succeeded())
+      dependsOn: build
+      jobs:
+      - deployment: deploy_hidi
+        dependsOn: []
+        environment: nuget-org
+        strategy:
+          runOnce:
+            deploy:
+              pool:
+                vmImage: ubuntu-latest
+              steps:
+              - task: DownloadPipelineArtifact@2
+                displayName: Download nupkg from artifacts
+                inputs:
+                  artifact: Nugets
+                  source: current
+              - task: DownloadPipelineArtifact@2
+                displayName: Download hidi executable from artifacts
+                inputs:
+                  source: current
+              - pwsh: |
+                  $artifactName = Get-ChildItem -Path $(Pipeline.Workspace)\Nugets -Filter Microsoft.OpenApi.*.nupkg -recurse | select -First 1
+                  $artifactVersion= $artifactName.Name -replace "Microsoft.OpenApi.", "" -replace ".nupkg", ""
+                  #Set Variable $artifactName and $artifactVersion
+                  Write-Host "##vso[task.setvariable variable=artifactVersion; isSecret=false; isOutput=true]$artifactVersion"
+                  Write-Host "##vso[task.setvariable variable=artifactName; isSecret=false; isOutput=true]$artifactName.FullName"
+                  echo "$artifactName"
+                  echo "$artifactVersion"
+                displayName: 'Fetch Artifact Name'
+              - task: 1ES.PublishNuget@1
+                displayName: 'NuGet push'
+                inputs:
+                  useDotNetTask: true
+                  packagesToPush: '$(Pipeline.Workspace)/Nugets/Microsoft.OpenApi.Hidi.*.nupkg'
```

Comment:
I'm not sure if there's a way to confirm but this may need to be double checked
```suggestion
                  packagesToPush: '$(Pipeline.Workspace)/Microsoft.OpenApi.Hidi.*.nupkg'
```

Reply:
I see that Hidi's `.nupkg` is still published to the Nugets folder in the artifacts section so I'm not sure whether this should change:
![image](https://github.com/microsoft/OpenAPI.NET/assets/36787645/d4eabf5e-6fc7-4f8f-a871-e99854f9727e)

Reply:
Let's resolve this for now. The issue i saw sometime back involved the `DownloadPipelineArtifact` task extracting the files in the `(Pipeline.Workspace)` and not the `(Pipeline.Workspace)\Nugets` folder. We can create a second PR if this breaks.